packages/doctype-tile/src/tile-doctype.ts

```diff
@@ -27,45 +30,130 @@ export interface TileParams extends DocParams {
     content?: Record<string, unknown>;
 }
 
+export interface TileMetadata {
+  controllers?: Array<string>
+  family?: string
+  schema?: CommitID | string
+  tags?: Array<string>
+  deterministic?: boolean
+}
+
+const headerFromMetadata = function(metadata?: TileMetadata | DocMetadata): CommitHeader {
+    if (typeof metadata?.schema === 'string') {
+        try {
+            CommitID.fromString(metadata.schema)
+        } catch {
+            throw new Error("Schema must be a CommitID")
+        }
+    }
+
+    const header: CommitHeader = {
+        controllers: metadata?.controllers,
+        family: metadata?.family,
+        schema: metadata?.schema?.toString(),
+        tags: metadata?.tags,
+    }
+
+    // Delete undefined keys from header
+    Object.keys(header).forEach(key => header[key] === undefined && delete header[key])
+    return header
+}
+
+const throwReadOnlyError = async (): Promise<void> => {
+    throw new Error('Historical document commits cannot be modified. Load the document without specifying a commit to make updates.')
+}
+
 /**
  * Tile doctype implementation
  */
 @DoctypeStatic<DoctypeConstructor<TileDoctype>>()
-export class TileDoctype extends Doctype {
+export class TileDoctype<T = Record<string, any>> extends Doctype {
+
+    static DOCTYPE_NAME = 'tile'
+    static DOCTYPE_ID = 0
 
     /**
-     * Change existing Tile doctype
-     * @param params - Change parameters
-     * @param opts - Initialization options
+     * Creates a Tile document.
+     * @param ceramic - Instance of CeramicAPI used to communicate with the Ceramic network
+     * @param content - Genesis contents. If 'null', then no signature is required to make the genesis commit
+     * @param metadata - Genesis metadata
+     * @param opts - Additional options
+     * TODO: Better way to specify "no content"? Should 'null' get converted to 'undefined'? What about the empty document {}?
```

Comment:
In the old API "no content" was expressed by just leaving the `content` field out of the `DocParams` entirely.  Now that `content` is a top-level argument, we need to decide how to represent "no content" especially as it pertains to the rule that genesis commits with no content don't need to be signed.  The version I went with here accepts both `undefined` and `null` as meaning "no content" and will ensure that the `data` field doesn't show up in the `GenesisCommit` in either case.  We could consider adding `{}` as an option for "no content", making `{}` the *only* way to specify "no content" (so that the `content` arg is always an object), whether we should include the `data` field in the GenesisCommit as an empty object, and any combination of the above.

But I think what I did here is likely fine.

Reply:
Thanks for clarifying. This seems good.